File: Fines2App/Values/Disposicion.cs
﻿using SqlOrganize;
using Utils;

namespace Fines2App.Values
{
    class Disposicion : EntityValues
    {
        public Disposicion(Db _db, string _entity_name, string? _field_id) : base(_db, _entity_name, _field_id)
        {
        }

        public override string ToString()
        {
            var s = "";

            EntityValues? v = ValuesTree("asignatura");
            if(v != null)
            {
                s += v.GetOrNull("nombre")?.ToString() ?? "?";
                s += " ";
                s += v.GetOrNull("codigo")?.ToString() ?? "?";
                s += " ";
            }

            v = ValuesTree("planificacion");
            if (v != null)
            {
                s += v.GetOrNull("anio")?.ToString() ?? "?";
                s += "/";
                s += v.GetOrNull("semestre")?.ToString() ?? "?";
            }

            return s.Trim();
        }
    }
}

    

File: src/Fines2/Model/Disposicion_.php
<?php

namespace Fines2\Model;

use \Fines2\Model\Disposicion;

use SqlOrganize\Sql\Entity;
use SqlOrganize\Utils\ValueTypesUtils;

use Exception;
use DateTime;

class Disposicion_ extends Disposicion
{
    public function getLabel(): string {
        return ($this->asignatura_?->nombre ?? "?") . " " . 
            ($this->asignatura_?->codigo ?? "?") . " " . 
            ($this->planificacion_?->anio ?? "?"). "/" .
            ($this->planificacion_?->semestre ?? "?") . " " .
            ($this->planificacion_?->plan_?->resolucion ?? "?") . " " . 
            ValueTypesUtils::acronym($this->planificacion_?->plan_?->orientacion ?? "") . " (" .
            ($this->horas_catedra ?? "?") . ")"; 
    }
}

